### DISCUSSION TITLE
powerdns panel not run after install ngnix

### DISCUSSION THREAD

**Discussion by famaserver** [ngoduykhanh/PowerDNS-Admin#1356] [General]
Hello
I have the same problem as this.
After installing pdns and its services, all services are properly active
But after the web server is working properly, the pdns panel is not loaded.
All services are active, including:
nginx
pdns
powerdns-admin.socket
powerdns-admin.service

But the admin panel does not open with port 9191.

**Comment by metrax** [comment]
Please provide us your nginx config files, your pda service file and logfiles of nginx and pda (only content depending on this problem) .

**Reply by famaserver** [reply]
> Please provide us your nginx config files, your pda service file and logfiles of nginx and pda (only content depending on this problem) .

I watched its installation tutorial from https://phoenixnap.com/kb/powerdns-ubuntu and followed it line by line.

As long as the Python web server is running, the panel will open, but when ngnix is installed, there is no way to access the panel, but if we connect to the IP server with port 80, the ngnix welcome page will appear.



ngnix:
```
server {
  listen *:80;
  server_name               localhost;

  index                     index.html index.htm index.php;
  root                      /opt/web/powerdns-admin;
  access_log                /var/log/nginx/powerdns-admin.local.access.log combined;
  error_log                 /var/log/nginx/powerdns-admin.local.error.log;

  client_max_body_size              10m;
  client_body_buffer_size           128k;
  proxy_redirect                    off;
  proxy_connect_timeout             90;
  proxy_send_timeout                90;
  proxy_read_timeout                90;
  proxy_buffers                     32 4k;
  proxy_buffer_size                 8k;
  proxy_set_header                  Host $host;
  proxy_set_header                  X-Real-IP $remote_addr;
  proxy_set_header                  X-Forwarded-For $proxy_add_x_forwarded_for;
  proxy_headers_hash_bucket_size    64;

  location ~ ^/static/  {
    include  /etc/nginx/mime.types;
    root /opt/web/powerdns-admin/powerdnsadmin;

    location ~*  \.(jpg|jpeg|png|gif)$ {
      expires 365d;
    }

    location ~* ^.+.(css|js)$ {
      expires 7d;
    }
  }

  location / {
    proxy_pass            http://unix:/run/powerdns-admin/socket;
    proxy_read_timeout    120;
    proxy_connect_timeout 120;
    proxy_redirect        off;
  }

}
```

**Reply by metrax** [reply]
You have to create an upstream:
```
upstream pda{
    server unix:/run/powerdns-admin/socket;
  }
```
  
and use it as proxy pass:
`proxy_pass            http://pda`

**Reply by famaserver** [reply]
> You have to create an upstream:
> 
> ```
> upstream pda{
>     server unix:/run/powerdns-admin/socket;
>   }
> ```
> 
> and use it as proxy pass: `proxy_pass http://pda`

Do you introduce training in this context?
Or is there a reference link or source of an article for this?

**Reply by metrax** [reply]
Just a part of my personal working config ;-)
